counsel.el

```diff
@@ -6505,8 +6533,11 @@ list as it may also be a build directory."
     (dolist (item counsel-compile-history)
       (let ((srcdir (get-text-property 0 'srcdir item))
             (blddir (get-text-property 0 'blddir item)))
-        (when (or (and srcdir (file-in-directory-p srcdir root))
-                  (and blddir (file-in-directory-p blddir root)))
+        (when (if counsel-compile-history-exact-src
+                  (let ((src (get-text-property 0 'src item)))
+                    (and src (string= src (buffer-file-name))))
```

Comment:
Nit: `(buffer-file-name)` is the same as `buffer-file-name`, here and elsewhere in this patch.

Reply:
No I user buffer-file-name directly throughout.